Patches and fixes for Android 12

svlad-90 opened the pull request:
Change vis parameter name to comply with Android 12
Change 'persist.vis.uri' to 'persist.vendor.vis.uri' in order to fit to the Android 12 properties naming convention

Reviewed-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>
Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

----

Add check to not send an empty location
Reviewed-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>
Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

----

Remove waiting for the connection to the VIS server
Remove waiting for the connection to the VIS server in order to not
block the HIDL client for too long

Reviewed-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>
Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

----

Refactor 'VisDataProvider'
Refactor 'VisDataProvider'. Remove redundant source code. Add
synchronization.

Reviewed-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>
Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

arminn commented:
Regarding commit : [EPMDAEPRAD-106] Changed 'persist.vis.uri' parameter name 
plz follow https://github.com/xen-troops/android_device_xenvm/pull/77

Regarding  [EPMDAEPRAD-106] Integrate 'android_vendor_xen_gnss' :
- Refactored 'VisDataProvider'. Removed redundant code. Added
synchronization.
- Added check to not send empty location
- Removed waiting for the connection to the VIS server, to not block the
client for too long

This looks like 3 unique patches could you please split it ? (and also obey the style I've provided in 77).
Also plz do not add additional formatting which is not the purpose of the patch.

Commit:
Change vis parameter name to comply with Android 12

Change 'persist.vis.uri' to 'persist.vendor.vis.uri' in order to fit
to the Android 12 properties naming convention

Reviewed-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>
Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

Commit:
Add check to not send an empty location

Reviewed-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>
Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

Commit:
Remove waiting for the connection to the VIS server

Remove waiting for the connection to the VIS server in order to not
block the HIDL client for too long

Reviewed-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>
Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

Commit:
Refactor 'VisDataProvider'

Refactor 'VisDataProvider'. Remove redundant source code. Add
synchronization.

Reviewed-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>
Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

svlad-90 commented:
Hi @arminn, done. Please, proceed with the review.
